src/core/lib/security/transport/security_handshaker.c

```diff
@@ -321,18 +341,9 @@ static void on_handshake_data_sent_to_peer(grpc_exec_ctx *exec_ctx, void *arg,
     security_handshaker_unref(exec_ctx, h);
     return;
   }
-  /* We may be done. */
-  if (tsi_handshaker_is_in_progress(h->handshaker)) {
+  if (h->handshaker_result == NULL) {
```

Comment:
Please add a comment explaining the connection between `h->handshaker_result` and whether or not we need to read from the peer.